### PR TITLE
Define enum types to allow cross-referencing

### PR DESCRIPTION
**Pull request by apasel422**
***
<a href="https://pr-preview.s3.amazonaws.com/apasel422/attribution-reporting-api/pull/630.html" title="Last updated on Nov 23, 2022, 1:46 PM UTC (23aabe1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/630/73acb0e...apasel422:23aabe1.html" title="Last updated on Nov 23, 2022, 1:46 PM UTC (23aabe1)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -561,12 +583,18 @@ random delay to deliver an [=aggregatable report=].
...
-To <dfn>parse filter data</dfn> given a |value| and a [=string=]
+A <dfn>registration type</dfn> is one of the following:
+
+<ul dfn-for="registration type">
+<li>"<dfn><code>source</code></dfn>"
+<li>"<dfn><code>trigger</code></dfn>"
+</ul>
+
+To <dfn>parse filter data</dfn> given a |value| and a [=registration type=]
 <dfn for="parse filter data"><var>registrationType</var></dfn>:
```
Comment (linnan-github) [inline index.bs:594]: Do we still need to define this var with the explicit type?

Comment (apasel422) [reply index.bs:594]: Did we ever need to?

Comment (linnan-github) [reply index.bs:594]: Probably we can get rid of that?

File: index.bs [hunk 2]
```diff
@@ -1215,14 +1212,12 @@ To <dfn noexport>create an attribution trigger</dfn> given a [=string=]
...
-        |value|["`filters`"] and [=parse filter data/registrationType=] set to
-        "<code>[=registration type/trigger=]</code>".
+        |value|["`filters`"] and "<code>[=registration type/trigger=]</code>".
     1. If |filters| is null, return null.
 1. Let |negatedFilters| be a new [=filter map=].
 1. If |value|["`not_filters`"] exists:
     1. Set |negatedFilters| to the result of running [=parse filter data=] with
-        |value|["`not_filters`"] and [=parse filter data/registrationType=] set
-        to "<code>[=registration type/trigger=]</code>".
+        |value|["`not_filters`"] and to "<code>[=registration type/trigger=]</code>".
```
Comment (linnan-github) [inline index.bs:1220]:
```suggestion
        |value|["`not_filters`"] and "<code>[=registration type/trigger=]</code>".
```

**Review by linnan-github — APPROVED** [review]

**Pull request merged**